rfc/helm-import.md

```diff
@@ -0,0 +1,515 @@
+
+```typescript
+import { Construct } from 'constructs';
+import { App, Chart, ChartProps } from 'cdk8s';
+import { Airflow } from './imports/airflow';
+
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
+    super(scope, id, props);
+
+    new Airflow(this, 'airflow', {
+      values: {
+        fullnameOverride: 'Foo',
+      }
+    });
+  }
+}
+
+const app = new App();
+new MyChart(app, 'typescript-demo');
+app.synth();
+```
+
+> **Note:**
+>
+> * You would need `Helm` to be installed on your machine for using this feature. 
+> * We assume that you are authenticated to access the helm chart being mentioned in the url.
+> * If a helm chart contains a `values.schema.json` then the `values` properties within the construct properties for the chart would have specific types. For instance, in the prior example `fullnameOverride` is not of `any` type and instead is of `string` type since `airflow` chart has a `values.schema.json` file in it.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @iliapolo
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We have added a new feature to the `cdk8s import` command which allows users to import helm charts into their cdk8s app.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to use helm charts within your cdk8s app. This feature would allow you to add values for helm chart in a type safe manner if a `values.schema.json` file is present within the chart being used.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Our current support for including helm charts to cdk8s app adds churn for the user to setup the [Helm construct](https://cdk8s.io/docs/latest/basics/helm/) and pass in values to the construct. This feature would have two advantages over the current solution:
+1. It would make it easier for the user to setup a construct for helm charts. They would just need to run the `cdk8s import` command with a valid url and we would auto generate the construct for them.
+2. If a `values.schema.json` file is present in the helm chart that the user is referring, then our generated construct would have type safe values properties. This is not supported in Helm construct currently.
+
+### Why should we _not_ do this?
+
+As mentioned above, we do currently have a [solution]((https://cdk8s.io/docs/latest/basics/helm/)) for including helm charts in cdk8s app. The proposed feature is definitely an enhancement but would take up developers time and effort to implement and maintain.
+
+### What is the technical solution (design) of this feature?
+
+#### Helm chart url
+
+The following would be the format for a helm url being passed into the import command:
+
+```
+cdk8s import helm:<repo-url>/<chart-name>@<chart-version>
```

Comment:
Are you assuming that the chart name is the last element separated by a `/`? Is this always true? Is this what `helm pull` assumes?

Reply:
Instead of assuming it. I would just treat whatever is between `helm:` and `@` as the "thing" you pass on to `helm pull` - then to get the chart name, just look for the Chart.yaml file in the output of `helm pull`.

Reply:
Hey, I think this `helm:<repo-url>/<chart-name>@<chart-version>` should be our syntax for providing a url. The reason for this being, 
* `repo-url` would be the chart repository url, like `https://charts.bitnami.com/bitnami` and this in `helm pull` is passed in with the `--repo` flag.
* `chart-name` is used to get a particular chart from the repo. 
For instance,
```
helm pull mysql --repo https://charts.bitnami.com/bitnami/
```

------

`helm pull` either accepts `absolute URLs(https://charts.bitnami.com/bitnami/mysql-9.10.10.tgz)` or `repo/chartname`(which is what we have after adding the repo like [this](https://artifacthub.io/packages/helm/bitnami/mysql?modal=install)). 

I can run the `helm repo add` for adding the chart and then pulling it, but I was avoiding it so that `helm pull` does not change anything locally for the user. 


-----

Do you think `helm:<repo-url>/<chart-name>@<chart-version>` could be restricting since we do not know what urls are possible?

Reply:
Ohh, I see we do callout, `helm repo add` with our current helm implementation. We can do the same for this. WDYT?

Link: https://github.com/cdk8s-team/cdk8s/blob/master/docs/basics/helm.md?plain=1#L14

Reply:
Synced offline and we are moving forward with `helm:<repo-url>/<chart-name>@<chart-version>` syntax.